### DISCUSSION TITLE
TypeError: value.charCodeAt is not a function with tailwind and PostCSS event after followed the tutorial

### DISCUSSION THREAD

**Discussion by Ridermansb** [tailwindlabs/tailwindcss#3449] [Help]
I already follow [the tutorial][1] and check the [discuesions tab][2] .

Now I'm receiving this error

```
ERROR in ./src/assets/app.css (./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].use[1]!./node_modules/postcss-loader/dist/cjs.js!./src/assets/app.css)
Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
TypeError: value.charCodeAt is not a function

```

[1]: https://tailwindcss.com/docs/installation#post-css-7-compatibility-build
[2]: https://github.com/tailwindlabs/tailwindcss/discussions/2929

**Comment by simonswiss** [comment]
Hey! 👋

Can you please share a reproduction of this error in a GitHub repo so it's a little easier to help?

**Comment by Ridermansb** [comment]
Hi @simonswiss , I fixed the error and I don't know how :) .
I think I removed `node_modules` and `package-lock.json` and install everything again.

**Comment by max-bruhn** [comment]
I am getting the same error. using these versions:

` 
"devDependencies": {
    "@tailwindcss/postcss7-compat": "^2.0.3",
    "autoprefixer": "^9.8.6",
    "postcss": "^7.0.35",
    "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.0.3"
  }
`

Error Message:
`
@tailwindcss/postcss7-compat 2.0.3

   🚀 Building: src/tailwind.css
  
   🚫 TypeError: value.charCodeAt is not a function
`

tailwind.css
`@tailwind base;
h1 {
  @apply text-4xl sm:text-6xl lg:text-7xl font-black 2xl:text-8xl;
}
h2 {
  @apply text-3xl sm:text-xl lg:text-3xl font-black 2xl:text-4xl;
}
h3 {
  @apply text-xl;
}
a {
  @apply text-accent;
}
strong {
  @apply text-accent font-bold;
}
@tailwind components;
@tailwind utilities;
`

If I have a blank tailwind.css file then it works. as soon as i add one of those lines @tailwind etc. it crashes. I tried all of them individually and every combination. 
I tried reinstalling several times.
Latest versions are throwing the other error: Error: PostCSS plugin tailwindcss requires PostCSS 8.

**Reply by simonswiss** [reply]
Possible to put this in a small public GitHub repo to try and replicate the error?

**Reply by max-bruhn** [reply]
i did manage to get it running by now, following this guide: https://dev.to/thefierycoder/setting-up-tailwind-css-v2-0-with-create-react-app-44h9

basically specifying the tailwind config file inside postcss config like so:
`module.exports = {
  plugins: {
    tailwindcss: { config: './tailwindcss-config.js' },
    autoprefixer: {},
  },
}`

**Comment by jerryharrison** [comment]
If anyone is using Tailwind UI I have the solution: Had similar problem with a similar setup as above.

https://tailwindui.com/changes-for-v2#updating-your-tailwind-ui-projects

Essentially when you're upgrading from v1 to v2 of Tailwind, you need to remove the require for tailwind/ui in your tailwind.config file.

Hope this helps someone!

**Reply by princejohnsantillan** [reply]
As a TailwindUI user this worked for me. Thanks @jerryharrison

**Reply by khatau** [reply]
Worked for me. Laravel breeze tailwind ui preset.

**Comment by animeshsinghweb** [comment]
Mine was solved by using the previous version of tailwind and postcss
```
  "devDependencies": {
    "autoprefixer": "9.8.0",
    "cross-env": "7.0.2",
    "cssnano": "4.1.10",
    "postcss-cli": "7.1.1",
    "release-it": "13.6.4",
    "tailwindcss": "^1.9.6"
  },
```
